dpnp/dpnp_iface_statistics.py

```diff
@@ -276,21 +269,17 @@ def average(a, axis=None, weights=None, returned=False, *, keepdims=False):
     """
 
     dpnp.check_supported_arrays_type(a)
+    usm_type, exec_q = get_usm_allocations([a, weights])
+
     if weights is None:
         avg = dpnp.mean(a, axis=axis, keepdims=keepdims)
         scl = dpnp.asanyarray(
             avg.dtype.type(a.size / avg.size),
-            usm_type=a.usm_type,
-            sycl_queue=a.sycl_queue,
+            usm_type=usm_type,
+            sycl_queue=exec_q,
         )
     else:
-        if not isinstance(weights, (dpnp_array, dpt.usm_ndarray)):
-            wgt = dpnp.asanyarray(
-                weights, usm_type=a.usm_type, sycl_queue=a.sycl_queue
-            )
-        else:
-            get_usm_allocations([a, weights])
-            wgt = weights
+        wgt = dpnp.asanyarray(weights, usm_type=usm_type, sycl_queue=exec_q)
```

Comment:
Wouldn't it be better to call `asanyarray` only when it is needed? (to avoid extra work)
```suggestion
        if not dpnp.is_supported_array_type(weights):
            weights= dpnp.asanyarray(weights, usm_type=usm_type, sycl_queue=exec_q)
```